src/net/client/xfr.rs

```diff
@@ -0,0 +1,1123 @@
+//! Parsing of AXFR/IXFR response messages for higher level processing.
+//!
+//! This module provides [`XfrResponseProcessor`] which enables you to process
+//! one or more AXFR/IXFR response messages in terms of the high level
+//! [`XfrEvent`]s that they represent without having to deal with the
+//! AXFR/IXFR protocol details.
+use std::fmt::Debug;
+
+use bytes::Bytes;
+use tracing::trace;
+
+use crate::base::iana::Opcode;
+use crate::base::wire::ParseError;
+use crate::base::{
+    Message, ParsedName, Record, RecordSection, Rtype, Serial,
+};
+use crate::rdata::{Soa, ZoneRecordData};
+
+//------------ XfrRecord ------------------------------------------------------
+
+/// The type of record processed by [`XfrResponseProcessor`].
+pub type XfrRecord =
+    Record<ParsedName<Bytes>, ZoneRecordData<Bytes, ParsedName<Bytes>>>;
+
+//------------ XfrResponseProcessor -------------------------------------------
+
+/// An AXFR/IXFR response processor.
+///
+/// [`XfrResponseProcessor`] can be invoked on one ore more sequentially
+/// AXFR/IXFR received response messages to verify them and during processing
+/// emit events which an implementor of [`XfrEventHandler`] can handle.
+pub struct XfrResponseProcessor<T: XfrEventHandler> {
+    evt_handler: T,
+
+    mode: Mode,
+}
+
+impl<T: XfrEventHandler> XfrResponseProcessor<T> {
+    /// Create a new XFR response processor.
+    ///
+    /// Events will be emitted to the given [`XfrEventHandler`] implementation.
+    pub fn new(evt_handler: T) -> Self {
+        Self {
+            evt_handler,
+            mode: Mode::default(),
+        }
+    }
+
+    /// Process a single AXFR/IXFR response message.
+    ///
+    /// During processing events will be emitted to the registered
+    /// [`XfrEventHandler`] for handling.
+    ///
+    /// Returns Ok(true) if the XFR response was the last in the seqence,
+    /// Ok(false) if more XFR response messages are needed to complete the
+    /// sequence, or Err on error.
+    pub async fn answer(
```

Comment:
Perhaps the method should be named `process_answer` or similar? The current name sounds like this method answers something which is very confusing for the client …

Reply:
I agree, but for consistency it was named after the existing `ClientSequence::answer()`. I'm not sure about the word "process" in the name or in the name of the type either, need to think about the exact API and naming to offer.